collect_app/src/main/res/values/strings.xml

```diff
@@ -689,4 +689,5 @@
     <string name="read_details">Read details</string>
     <string name="google_sheets_encrypted_message">Encrypted forms can\'t be submitted to Google Sheets.</string>
     <string name="number_picker_title">Number Picker</string>
+    <string name="too_complex_form">The form is too complex and cannot be parsed on this device. Try simplifying used expressions.</string>
```

Comment:
Maybe "used" is not necessary in "used expressions"? Would it be worth linking to a forum post or documentation about what kinds of complex expressions might cause this kind of problem?

Reply:
I don't think any link is needed here. We don't have any doc about this case and since each case is different the forum topic https://forum.opendatakit.org/t/crash-when-opening-a-complex-form-on-android-4/14264 might be not really helpful.
What about a message like:
`The form is too complex and cannot be parsed on this device. Try simplifying expressions or ask for help on the forum.`

Reply:
That makes sense. And like you said in the PR description, it's rare anyway. I like your second sentence. I think in general, mentioning the forum has been avoided for user-facing strings but in this case the person designing the form would probably see it first. Maybe the first sentence can be simplified to `This form is too complex for this device.`?

Reply:
Done.